### PR TITLE
Use execnet main_thread_only execmodel

### PR DESCRIPTION
**Pull request by zmedico**
Use the execnet `main_thread_only `execmodel so that code which expects to run in the main thread will just work.  This execmodel has been merged to the execnet master branch via pytest-dev/execnet#243, so this patch should not be merged until there is a released version of execnet supporting the `main_thread_only` execmodel.

Also increase minimum python version to 3.8 since execnet dropped 3.7 support in pytest-dev/execnet#245.

Closes: pytest-dev/pytest-xdist#620

### CODE REVIEW

**Review by RonnyPfannschmidt — APPROVED** [review]
As gnarly as required by the hack, thanks

File: pyproject.toml
Comment (nicoddemus) [inline pyproject.toml:36]: Once a new execnet is released:

```suggestion
    "execnet>=2.1.0",
```

File: changelog/620.bugfix
Comment (nicoddemus) [inline changelog/620.bugfix:3]:
```suggestion
Use the ``execnet`` new ``main_thread_only`` "execmodel" so that code which expects to only run in the main thread will now work as expected.
```

Let's mention the Python 3.7 drop into its own note. :+1:

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Great work, thanks @zmedico!

Leaving this as "Request changes" to make it clear this is waiting on a new execnet release. :+1:

**Comment by bluetech** [comment]
@zmedico we now depend on execnet>=2.1.0 in master so I think this unblocked. @zmedico can you rebase? Note that master already dropped py3.7 as well so you can remove these changes from the PR changelog notes.

**Comment by bluetech** [comment]
@zmedico The just-merged typing PR created other conflicts (sorry about that, I forgot). Can you rebase again? (if you want I can do it as well).

@nicoddemus You marked "requested changes" but now that we depend on new execnet I assume you're good with merging?

After this is merged I can do the 3.6.0 release.

**Review by nicoddemus — APPROVED** [review]
Thanks @zmedico!

**Comment by nicoddemus** [comment]
> After this is merged I can do the 3.6.0 release.

That would be great, thanks @bluetech!

**Comment by webknjaz** [comment]
xref https://github.com/pytest-dev/pytest-xdist/pull/1070#issuecomment-2067290245

**Pull request merged**
